cmd/kubelet/app/server.go

```diff
@@ -783,6 +793,9 @@ func makePodSourceConfig(kc *KubeletConfig) *config.PodConfig {
 type KubeletConfig struct {
 	Address                        net.IP
 	AllowPrivileged                bool
+	Authenticator                  authenticator.Request
+	AuthorizerAttributeGetter      authorizer.RequestAttributesGetter
+	Authorizer                     authorizer.Authorizer
```

Comment:
yeah, that makes it cleaner. updated to a single interface field. that also let me remove the special `nil` behavior in KubeletAuth... if you want no auth, leave KubeletConfig.Auth nil. If you want auth, set it to a NewKubeletAuth() with non-nil components